pkg/validation/internal/removed_apis.go

```diff
@@ -335,25 +336,56 @@ func getRemovedAPIsOn1_25From(bundle *manifests.Bundle) (map[string][]string, ma
 		}
 	}
 
-	warnIfDeprecated := func(res string, msg string) {
-		switch res {
-		case "cronjobs":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "endpointslices":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "events":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "horizontalpodautoscalers":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "poddisruptionbudgets":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "podsecuritypolicies":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "runtimeclasses":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
+	warnIfDeprecated := func(gr schema.GroupResource, msg string) {
+		switch gr.Group {
+		case "batch":
+			if gr.Resource == "cronjobs" {
+				warnDeprecatedAPIs[gr.Resource] = append(warnDeprecatedAPIs[gr.Resource], msg)
+			}
+		case "discovery.k8s.io":
+			if gr.Resource == "endpointslices" {
+				warnDeprecatedAPIs[gr.Resource] = append(warnDeprecatedAPIs[gr.Resource], msg)
+			}
+		case "events.k8s.io":
+			if gr.Resource == "events" {
+				warnDeprecatedAPIs[gr.Resource] = append(warnDeprecatedAPIs[gr.Resource], msg)
+			}
+		case "autoscaling":
+			if gr.Resource == "horizontalpodautoscalers" {
+				warnDeprecatedAPIs[gr.Resource] = append(warnDeprecatedAPIs[gr.Resource], msg)
+			}
+		case "policy":
+			if gr.Resource == "poddisruptionbudgets" || gr.Resource == "podsecuritypolicies" {
+				warnDeprecatedAPIs[gr.Resource] = append(warnDeprecatedAPIs[gr.Resource], msg)
+			}
+		case "node.k8s.io":
+			if gr.Resource == "runtimeclasses" {
+				warnDeprecatedAPIs[gr.Resource] = append(warnDeprecatedAPIs[gr.Resource], msg)
+			}
 		}
 	}
 
+	apiGroupHasDeprecation := func(group string) bool {
```

Comment:
Same question here. Can we make this a `map[string]struct{}` (or use `sets.String` from k8s.io/apimachinery)?

Reply:
After adding maps to resolve your previous comment, I ended up removing this function definition as the map used by `warnIfDeprecated` has the `schema.GroupResource` as a key. If the given `schema.GroupResource` doesn't match any of the keys it doesn't warn.

I think the only downside to having removed this function is that now it always runs the loop to create a `schema.GroupResource` and verify it instead of only if the group has a deprecated api. If this is something we are concerned about I can re-implement this function.

@joelanford @oceanc80 WDYT?